app/models/user.rb

```diff
@@ -7,12 +7,22 @@ class User < ApplicationRecord
   validates :email, presence: true, uniqueness: true
   normalizes :email, with: ->(email) { email.strip.downcase }
 
+  normalizes :first_name, :last_name, with: ->(value) { value.strip.presence }
+
   enum :role, { member: "member", admin: "admin" }, validate: true
 
   generates_token_for :password_reset, expires_in: 15.minutes do
     password_salt&.last(10)
   end
 
+  def display_name
+    [ first_name, last_name ].compact.join(" ").presence || email
```

Comment:
This falls back to email because is also used on the list of members, and it looks odd without some sort of identifier. I've added a screenshot below for reference.